src/js/controllers/tab-receive.js

```diff
@@ -88,9 +82,9 @@ angular.module('copayApp.controllers').controller('tabReceiveController', functi
     });
   };
 
-  $scope.setWallet = function(index) {
-    $scope.wallet = $scope.wallets[index];
-    $scope.walletIndex = index;
+  $scope.setWallet = function(wallet) {
```

Comment:
This looks better but the view is still using this function passing the wallet index: https://github.com/bitpay/copay/blob/master/www/views/tab-receive.html#L80